### PR TITLE
[AppSec] Report attacks with Trace Span data

### PR DESCRIPTION
**Pull request by bantonsson**
This is the initial version of sending the attack report payload together with the trace span data. Most changes in this PR relate to changes in the serialized JSON of the AppSec attack report, as well as the removal of the old reporting code. This PR also introduce the concept of a `TraceSegment` where tags can be set at the _segment_ level (currently at the local root span of the segment).

### CODE REVIEW

**Review by devinsba — APPROVED** [review]
I'm still not sure about the TraceSegment API, it feel polluted to me to have the intent of "stable reference to the local root" but also have methods that modify a "current" span. Maybe we should mark it in some way as unstable so that we can iterate on it without some customer accidentally depending on it

File: dd-java-agent/appsec/src/main/schemas/raw/events/app-sec-event-1.0.0.json [hunk 1]
```diff
@@ -0,0 +1,113 @@
+{
```
[the remark this answers is not in the document]
Comment (bantonsson) [reply dd-java-agent/appsec/src/main/schemas/raw/events/app-sec-event-1.0.0.json:1]: https://github.com/DataDog/libddwaf/blob/153890d99d48bb408533e2cabbc0da67e1ee1989/schema/appsec-event-1.0.0.json

[the remark this answers is not in the document]
Comment (bantonsson) [reply dd-java-agent/appsec/src/main/schemas/raw/events/app-sec-event-1.0.0.json:1]: I don't think that it's been tagged properly.

**Comment by bantonsson** [comment]
I agree that it would be nice to _hide_ the `TraceSegment` for now. Maybe just move it to a package named `...api.internal.`?

As for the polluted part, I think that we might see the role of the `TraceSegment` a bit differently. Right now its main purpose is to hide the `Span`, and its surface are the size of a small South American country, from AppSec. This is to ensure that AppSec keeps consuming the data via the `InstrumentationGateway`, and also doesn't have to concern itself with the inner workings of local root spans, and the like.

File: dd-java-agent/appsec/src/main/schemas/raw/events/app-sec-event-1.0.0.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (ValentinZakharov) [reply dd-java-agent/appsec/src/main/schemas/raw/events/app-sec-event-1.0.0.json:1]: I copied schema from the WAF's repo
https://github.com/DataDog/libddwaf/blob/master/schema/appsec-event-1.0.0.json
Do you have any recommendations regarding to versioning/tagging?

**Pull request merged**